src/chen.jl

```diff
@@ -1,25 +1,32 @@
-
-using LinearAlgebra
-
 """
     RaoBlackwellLedoitWolf(shrinkage)
 
 Rao-Blackwell theorem Ledoit-Wolf covariance estimator. The parameter
-`shrinkage` is either equal to `:auto` and optimal shrinkage is calculated,
+`shrinkage` is either equal to `:optimal` and optimal shrinkage is calculated,
 or it is a number between 0 and 1.
 """
 struct RaoBlackwellLedoitWolf{S<:Union{Symbol, Real}} <: CovarianceEstimator
     shrinkage::S
+    function RaoBlackwellLedoitWolf(s::T) where T<:Real
+        @assert 0 ≤ s ≤ 1 "Shrinkage value should be between 0 and 1"
+        new{T}(s)
+    end
+    function RaoBlackwellLedoitWolf(s::Symbol=:optimal)
+        @assert s ∈ [:optimal] "Shrinkage setting not supported"
+        new{Symbol}(s)
+    end
 end
 
-RaoBlackwellLedoitWolf() = RaoBlackwellLedoitWolf{Symbol}(:auto)
-
-function chenshrinkagetarget(X::AbstractMatrix{<:Real}; dims=2)
-    p = size(X, dims)
-    C = cov(X; dims=dims)
-    (tr(C)/p) * one(C)
+function rblw_optimalshrinkage(Ŝ::AbstractMatrix, n::Int, p::Int)
```

Comment:
I'm just cautious about Unicode. Sometimes I see combining characters that are incorrectly rendered (that's why I've written `hat` instead of adding a hat in a few identifiers). Greek letters and Unicode operators look relatively harmless though.